src/terminal/parser/ft_fuzzer/string_helper.h

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#ifndef __STRING_HELPER_H__
+#define __STRING_HELPER_H__
+
+#pragma once
+
+void AppendFormat(
+    _In_ std::string& to,
+    _In_z_ _Printf_format_string_ const char* format,
+    ...)
+{
+    THROW_HR_IF_NULL(E_INVALIDARG, format);
+
+    va_list args;
+    va_start(args, format);
+
+    const auto currentLength = to.length();
+    const auto appendLength = _vscprintf(format, args);
+
+    THROW_HR_IF(E_FAIL, appendLength < 0);
+
+    to.resize(currentLength + appendLength + 1);
+
+    const auto len = vsprintf_s(to.data() + currentLength, appendLength + 1, format, args);
+    THROW_HR_IF(E_FAIL, len < 0);
+
+    to.resize(currentLength + appendLength);

```

Comment:
I'm trying to carefully enough not to introduce bugs.
According to docs, [_vscprintf](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/vscprintf-vscprintf-l-vscwprintf-vscwprintf-l?view=vs-2019) `The value returned does not include the terminating null character`.
And [sprintf_s](https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/sprintf-s-sprintf-s-l-swprintf-s-swprintf-s-l?view=vs-2019) `guarantees that the buffer will be null-terminated`.
I allocates one more byte for null character and then remove it.

Also `CString::AppendFormat` uses exception so I uses too.

Reply:
That's fine, no problem.